config/routes.rb

```diff
@@ -1,9 +1,21 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-   root "users#index"
-  # Defines the root path route ("/")
-  # root "articles#index"
+  root "users#index", as: :start
+
+  # Users
   get 'users/:id' => 'users#show', as: :user_id
+
+  # Posts
   get 'users/:user_id/posts' => 'posts#index'
+  get 'users/:user_id/posts/new' => 'posts#new', as: :post_new
+  post 'users/:user_id/posts/new' => 'posts#create', as: :post
```

Comment:
- [ ] for the creating posts id you don't need user_id as a param, you are using current_user from ``application_controller.rb`` , kindly remove the user id param from :
```Ruby
get 'users/:user_id/posts/new' => 'posts#new', as: :post_new
post 'users/:user_id/posts/new' => 'posts#create', as: :post
```
to prevent access other user's forms